knimin/templates/pm_plate_list.html

```diff
@@ -0,0 +1,156 @@
+<style>
+  p {
+    font-weight: bold;
+  }
+  a {
+    color: black;
+  }
+  button {
+    padding: 5px 10px 5px 10px;
+    text-decoration: none;
+    font-size: 80%;
+  }
+  select {
+    border: 1px solid black;
+    padding: 5px 10px 5px 10px;
+    height: 25px;
+    margin-bottom: 1px;
+  }
+  button.inline {
+    height: 25px;
+    width: 25px;
+    padding: 2px 4px 2px 4px;
+    border: 0;
+    background-color: transparent;
+    text-decoration: none;
+    font-size: 80%;
+  }
+  button.inline:hover {
+    background-color: dimgrey;
+    color: white;
+  }
+  table {
+    border-top: 1px solid black;
+  }
+  table.dataTable thead th {
+    text-align: left;
+    padding: 5px 10px 5px 10px;
+  }
+  tbody.list tr td {
+    vertical-align: middle;
+    padding: 5px 10px 5px 10px;
+  }
+  tbody.list tr:hover {
+    background-color: lightgray;
+  }
+  tbody.list a:hover {
+    color: darkblue;
+    text-decoration: underline;
+  }
+  .dataTables_wrapper .dataTables_info {
+    font-size: 90%;
+  }
+  .dataTables_wrapper .dataTables_filter {
+    font-size: 90%;
+  }
+  .dataTables_wrapper .dataTables_length {
+    font-size: 90%;
+  }
+  .dataTables_wrapper .dataTables_paginate {
+    font-size: 90%;
+  }
+  .dataTables_wrapper .dataTables_paginate .paginate_button {
+    padding: 4px 8px 4px 8px;
+    text-decoration: none;
+  }
+</style>
+<script type="text/javascript">
+  $(document).ready(function(){
+    $('#plate_list').DataTable({
+      scrollY: 450,
+      scrollCollapse: true,
+      pageLength: 50
+    });
+    $('#plate_list tbody').on('click', 'tr', function(){
+        $(this).toggleClass('selected');
+    });
+  });
+</script>
+{% end %}
+
+{% block content %}
+<h3 id="page_title">{{target[0].upper() + target[1:]}} Plate List</h3>
+<p>
+  <button id="btnNew"><i class="fa fa-plus"></i> Create new</button>
+  <button id="btnDel"><i class="fa fa-remove"></i> Delete selected</button>
+  <button id="btnEdit"><i class="fa fa-edit"></i> Batch-edit selected</button>
+  <button id="btnDNA"><i class="fa fa-flask"></i> Convert to DNA plate</button>
+  <button id="btnEdit"><i class="fa fa-download"></i> Export list</button>
+</p>
+
+<table id="plate_list">
+  <thead>
+    <tr>
+      <th width="50px">ID</th>
+      <th width="225px">Name</th>
+      <th width="75px">Type</th>
+      <th width="50px">Filled</th>
+      <th width="50px">Study</th>
+      <th width="75px">Person</th>
+      <th width="75px">Date</th>
+      <th width="100px">Actions</th>
+    </tr>
+  </thead>
+  <tbody class="list">
+    {% for plate in plates %}
+    <tr>
+      <td>{{plate['id']}}</td>
+      <td><a href="/pm_plate_map/?target=sample&id={{plate['id']}}">{{plate['name']}}</a></td>
+      <td title="{{plate['type'][1]}}">{{plate['type'][0]}}</td>
+      <td>{{plate['count']}}
+        <span style="font-size:80%">
+          {% set x = plate['study'][0] / float(plate['type'][1]) %}
```

Comment:
plate['type'][1] is the total number of wells in that plate type. It won't be zero. I can do this calculation in handlers. I thought it may be better to let the webpage know the total number of wells, rather than a percentage. It makes no difference now, but maybe this information can be used in the future. What do you think?